src/handlers/genesis.rs

```diff
@@ -0,0 +1,80 @@
+use std::{str::FromStr, sync::Arc};
+
+use cardano::block;
+use exe_common::genesis_data;
+use iron;
+use iron::{IronResult, Request, Response};
+use iron::status;
+use router;
+use router::Router;
+
+use super::common;
+use super::super::config::Networks;
+
+pub struct Handler {
+    networks: Arc<Networks>,
+}
+impl Handler {
+    pub fn new(networks: Arc<Networks>) -> Self {
+        Handler { networks: networks }
+    }
+    pub fn route(self, router: &mut Router) -> &mut Router {
+        router.get(":network/genesis/:hash", self, "genesis_by_hash")
+    }
+}
+
+impl iron::Handler for Handler {
+    fn handle(&self, req: &mut Request) -> IronResult<Response> {
+        let ref network_name = req
+            .extensions
+            .get::<router::Router>()
+            .unwrap()
+            .find("network")
+            .unwrap();
+
+        if !common::validate_network_name(network_name) {
+            return Ok(Response::with((status::BadRequest, "Invalid network")));
+        }
+
+        let _net = match self.networks.get(network_name.to_owned()) {
+            None => return Ok(Response::with((status::BadRequest, "Failed to parse network"))),
+            Some(net) => net,
+        };
+
+        let ref hash = req
```

Comment:
I can spot `req.extensions.get(..).unwrap()` twice (one for network, one for hash), maybe this need refactoring ?

Reply:
Makes sense! Extracted `params` into separate variable. Also made the same fix in `block` handler, cuz mostly copied code from there )